### PR TITLE
QA - middle

### PR DESCRIPTION
**Pull request by maciejhubert**
Added middle level which includes:
 - test automation basics
 - improving the process
 -  ~~git~~

### CODE REVIEW

File: content/qa/03_middle.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+## Areas
+
+**Competence**
+
+- understanding basic differences between manual and automated tests 
+- setting up test automation structure on a project
+- covering regression tests with end-to-end tests
+- using basic framework functions
+- creating useful custom functions in terms of test automation
+- using CSS selectors along with writing tests
```
Comment (PoslinskiNet) [inline content/qa/03_middle.md:12]: Consider applying our competence template, that is nicely rendered and gives a nice glimpse from high-level one sentence description, to more detailed list of competences

```
{{%bubble %}}

## Interview

**Points:** 2

**Description:** You can present yourself as a competent person in a given area

**Person which successfully completed requirement for given block can:** 

- Can identify competencies are required for the given job
- Can communicate well during the interview with the customer
- Can talk about previous experience in the given area
- Can explain how company works
- Can explain software development processes along with common best practices

{{% /bubble%}}
```

[Code](https://github.com/Selleo/DevPath/blob/master/content/web_development/skills/interview.md) and [preview](https://selleo.com/devpath/web_development/skills/interview/)

File: content/qa/03_middle.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
+# Middle
```
Comment (PoslinskiNet) [inline content/qa/03_middle.md:1]: Don't forget about the meta tag - see the example below.

File: content/qa/03_middle.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+- Set up Cypress or some other test automation framework from scratch
+- Write and run a basic test case (e.g. logging in and some navigation)
+- Write an example custom command (e.g. generating a unique ID, selecting filters)
+- Use CSS selectors to refer to proper elements on a website
+
+---
+
+## 📦 Improving the process
+
+Constantly monitor ongoing process and workflow on your proujects. Point out flaws and suggest changes if needed.
```
Comment (PoslinskiNet) [inline content/qa/03_middle.md:61]:
```suggestion
Constantly monitor ongoing process and workflow on your projects. Point out flaws and suggest changes if needed. 
```

File: content/qa/03_middle.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+### 🎤 Interview
+
+- What should be done if you think that some tasks need more attention on a project?
+- When is it best to create test scenarios e.g. in Gherkin? What are basic values of such scenarios?
+- In case there is no budget for a QA engineer on a project - what would you suggest to maintain basic standards of delivered software?
+- Are there any cases of a flaw in the process on your project? What should be done to mitigate the flaws?
+
+### 📝 Katas
+
+- Tell me what are some solutions of imperfect process, based on case studies described in the document. Maybe you have other ideas how to solve those problems?
```
Comment (PoslinskiNet) [inline content/qa/03_middle.md:76]:
```suggestion
- What are some solutions to the imperfect process, based on case studies described in the document. How to solve those problems?
```

File: content/qa/03_middle.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+
+---
+
+## 📦 Git
+
+Be aware of basic git commands used for navigating between branches and handling changes. An enhanced version about git can be found in Common section of devpath.
```
Comment (PoslinskiNet) [inline content/qa/03_middle.md:83]: You can point it via link.

File: content/qa/03_middle.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+- Are there any cases of a flaw in the process on your project? What should be done to mitigate the flaws?
+
+### 📝 Katas
+
+- Tell me what are some solutions of imperfect process, based on case studies described in the document. Maybe you have other ideas how to solve those problems?
+
+
+---
+
+## 📦 Git
```
Comment (PoslinskiNet) [inline content/qa/03_middle.md:81]: How junior/independent are dealing with git on the projects now?

**Review by PoslinskiNet — CHANGES_REQUESTED** [review]
@maciejhubert I've left small feedback.

File: content/qa/03_middle.md (hunk 5 above)
Comment (maciejhubert) [reply content/qa/03_middle.md]: done

File: content/qa/03_middle.md (hunk 4 above)
Comment (maciejhubert) [reply content/qa/03_middle.md]: done

File: content/qa/03_middle.md (hunk 3 above)
Comment (maciejhubert) [reply content/qa/03_middle.md]: done

File: content/qa/03_middle.md (hunk 2 above)
Comment (maciejhubert) [reply content/qa/03_middle.md:3]: If you mean this:
`+++ title = "Middle" +++`
then it is done

**Comment by PoslinskiNet** [comment]
@maciejhubert something is not ok - build is failing due to:

image

File: content/qa/03_middle.md (hunk 1 above)
Comment (maciejhubert) [reply content/qa/03_middle.md]: done

File: content/qa/03_middle.md (hunk 6 above)
Comment (maciejhubert) [reply content/qa/03_middle.md]: As these are all basics of git, I have moved whole section to independent

**Comment by maciejhubert** [comment]
@PoslinskiNet, feedback applied. I will update Independent with git section in another PR

**Pull request merged**